generate/generate_report_test.go

```diff
@@ -0,0 +1,265 @@
+// Copyright 2022 Mineiros GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package generate_test
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/mineiros-io/terramate/generate"
+)
+
+func TestReportRepresentation(t *testing.T) {
+	type testcase struct {
+		name   string
+		report generate.Report
+		want   string
+	}
+
+	tcases := []testcase{
+		{
+			name:   "empty report",
+			report: generate.Report{},
+			want:   "Nothing to do, code generation is updated",
+		},
+		{
+			name: "with bootstrap err",
+			report: generate.Report{
+				BootstrapErr: errors.New("such fail, much terrible"),
+			},
+			want: `Fatal failure preparing for code generation.
```

Comment:
commented here: https://github.com/mineiros-io/terramate/pull/223#discussion_r811856276
Not so sure if they are fragile, you are validating the string representation of a type, if the string representation of a type changes (it is something public/exported to the client) then the test fails, correctly, because something public changed on the API. It felt odd to solve this only for simple errors (exporting things on the API that I don't see any use...but maybe there is) but then for the actual report we would have the same problem anyway, if the formatting of the report changes tests that validate the report will break, because..well..the report changed (tests not interested on the string representation of the report test things using the Report type itself, ignoring the string representation, and the ones that are interested on that, if it changes they should fail).